include/caf/click_detector.rst

```diff
@@ -0,0 +1,80 @@
+.. _caf_click_detector:
+
+CAF: Click detector module
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |click_detector| of the :ref:`lib_caf` (CAF) is responsible for generating a ``click_event`` when a known type of click is recorded for the button defined in the module configuration.
+
+Click type
+**********
+
+The module records the following click types:
+
+* :c:enumerator:`CLICK_SHORT` - Button pressed and released after short time.
+* :c:enumerator:`CLICK_NONE` - Button pressed and held for a period of time that is too long for :c:enumerator:`CLICK_SHORT`, but too short for :c:enumerator:`CLICK_LONG`.
+* :c:enumerator:`CLICK_LONG` - Button pressed and held for a long period of time.
+* :c:enumerator:`CLICK_DOUBLE` - Two sequences of the button press and release in a short time interval.
+
+The exact values of time intervals for click types are defined in the :file:`subsys/caf/modules/click_detector.c` file.
```

Comment:
Just a reflection here. @padu shouldn't we make those make those time intervals configurable? In particular short and long press. If not per button, at least globally.